extensions/mailer/runtime/src/main/java/io/quarkus/mailer/runtime/MailClientProducer.java

```diff
@@ -56,10 +60,90 @@ public void stop() {
+
         io.vertx.ext.mail.MailConfig cfg = toVertxMailConfig(config, tlsConfig);
         return MailClient.createShared(vertx, cfg);
     }
 
+    private io.vertx.ext.mail.DKIMSignOptions toVertxDkimSignOptions(DkimSignOptionsConfig optionsConfig) {
+        DKIMSignOptions vertxDkimOptions = new io.vertx.ext.mail.DKIMSignOptions();
+
+        if (optionsConfig.auid.isPresent()) {
+            vertxDkimOptions.setAuid(optionsConfig.auid.get());
+        }
+
+        if (optionsConfig.bodyCanonAlgo.isPresent()) {
+            vertxDkimOptions.setBodyCanonAlgo(CanonicalizationAlgorithm.valueOf(optionsConfig.bodyCanonAlgo.get().toString()));
+        }
+
+        if (optionsConfig.bodyLimit.isPresent()) {
+            int bodyLimit = optionsConfig.bodyLimit.getAsInt();
+            if (bodyLimit <= 0) {
+                throw new ConfigurationException("Body limit must be greater than zero.");
+            }
+            vertxDkimOptions.setBodyLimit(bodyLimit);
+        }
+
+        if (optionsConfig.expireTime.isPresent()) {
+            long expireTime = optionsConfig.expireTime.getAsLong();
+            if (expireTime <= 0) {
+                throw new ConfigurationException("Expirate time must be greater than zero.");
+            }
+            vertxDkimOptions.setExpireTime(expireTime);
+        }
+
+        if (optionsConfig.headerCanonAlgo.isPresent()) {
+            vertxDkimOptions.setBodyCanonAlgo(CanonicalizationAlgorithm.valueOf(optionsConfig.bodyCanonAlgo.get().name()));
+        }
+
+        if (optionsConfig.privateKey.isPresent()) {
+            vertxDkimOptions.setPrivateKey(optionsConfig.privateKey.get());
+        } else if (optionsConfig.privateKeyPath.isPresent()) {
+            vertxDkimOptions.setPrivateKeyPath(optionsConfig.privateKeyPath.get());
+        }
+
+        if (optionsConfig.sdid.isPresent()) {
+            vertxDkimOptions.setSdid(optionsConfig.sdid.get());
+        }
+
+        if (optionsConfig.selector.isPresent()) {
+            vertxDkimOptions.setSelector(optionsConfig.selector.get());
+        }
+
+        if (optionsConfig.signAlgo.isPresent()) {
+            DkimSignOptionsConfig.DkimSignAlgorithmOption signAlgoOption = optionsConfig.signAlgo.get();
+
+            DKIMSignAlgorithm selectedAlgo;
+            if (signAlgoOption.equals(DkimSignOptionsConfig.DkimSignAlgorithmOption.SHA1withRSA)) {
+                selectedAlgo = DKIMSignAlgorithm.RSA_SHA1;
+            } else {
+                selectedAlgo = DKIMSignAlgorithm.RSA_SHA256;
+            }
+            vertxDkimOptions.setSignAlgo(selectedAlgo);
+        }
+
+        if (optionsConfig.signatureTimestamp.isPresent()) {
+            vertxDkimOptions.setSignatureTimestamp(optionsConfig.signatureTimestamp.get());
+        }
+
+        if (optionsConfig.signedHeaders.isPresent()) {
+            List<String> headers = optionsConfig.signedHeaders.get()
```

Comment:
This shouldn't be needed anymore

Reply:
`ConfigItem` deals with blank strings? I didn't know.